### DISCUSSION TITLE
DelegateCommand.  Something like "ObservesCollection"?

### DISCUSSION THREAD

**Discussion by jmo-gelsight** [PrismLibrary/Prism#2526] [WPF]
I use `DelegateCommand.ObservesProperty` all the time.   But I have a case where I wish there was something different.   I'm wondering if there is but I just have missed it. 

It happens when my view-models expose a command whose enabled status depends upon the content of an `ObservableCollection` also exposed by the same viewmodel.  Some times these commands are only enabled when a certain number of items are in the collection.  Others, only when a certain types of items are in it.  

My typical workaround is to subscribe to `INotifyCollectionChanged.CollectionChanged ` myself and manually update the command's status.  Ideally, what I would like is for `DelegateCommand `to have a function called `ObservesCollection()`  that would trigger a check whenever the collection in question sent out a `CollectionChanged`.

I was about to try to derive my own class from `DelegateCommand` to add this but am I missing some obvious way to do this?

**Comment by AmrAlSayed0** [comment]
Observing `ObservesCollection.Count` could work but I don't think they will work for replace operations.

**Reply by jmo-gelsight** [reply]
Can I use ObservesProperty on a property that belongs to another object?  I was under the impression that I could only use ObservesProperty on my own properties, not on a property of a... property.  I've always just written it as

`ObservesProperty(() => SomePropertyOfMine)`

Because frankly, Count might be enough if I could do something like this:

`ObservesProperty(MyCollection => MyCollection.Count)`

**Reply by AmrAlSayed0** [reply]
Yes, Prism supports responding to nested properties. In fact, it responds to _all_ the properties in the property chain if each (or any) of these objects implement INPC interface.
`.ObserveProperty(() => obj.Property1.Property2.Property3)` would cause the `.CanExecute` to be revaluated on any change on any of the 3 properties.

**Reply by jmo-gelsight** [reply]
I have to admit, I feel awfully stupid for not realizing that I could do this.  I never understood that ObservesProperty would actually be triggered for properties "that far down the chain".  I will try this.

Still I might end up trying to implement my own ObservesCollection anyway, if only for sake of symmetry
